File: hidden-api/src/main/java/android/hardware/input/InputManagerHidden.java
package android.hardware.input;

import android.view.InputEvent;

import dev.rikka.tools.refine.RefineAs;

@RefineAs(InputManager.class)
public class InputManagerHidden {
    public boolean injectInputEvent(InputEvent event, int mode) {
        throw new RuntimeException("Stub!");
    }
}

File: hidden-api-contract/src/main/java/com/xaxaxax/moonclicker/hiddenapi/contracts/InputManagerHidden.kt
package com.xaxaxax.moonclicker.hiddenapi.contracts

internal val INPUT_MANAGER_HIDDEN = StubContracts(
    stubs = mapOf("android.hardware.input.InputManagerHidden" to "android.hardware.input.InputManager"),
    contracts = listOf(
        MemberContract(
            owner = "android.hardware.input.InputManager",
            member = MethodMember(
                name = "injectInputEvent",
                parameters = listOf("android.view.InputEvent", "int"),
                returns = "boolean",
            ),
        ),
    ),
)
